test_bst.py

```diff
@@ -266,3 +284,132 @@ def test_empty_tree_traversals(empty_bst):
     assert list(empty_bst.post_order()) == []
     assert list(empty_bst.in_order()) == []
     assert list(empty_bst.breadth_first()) == []
+
+
+def test_delete_empty(empty_bst):
+    empty_bst.delete(7)
+    assert empty_bst.root is None
+
+
+def test_delete_with_no_children(empty_bst):
+    empty_bst.insert(7)
+    assert empty_bst.root.value == 7
+    empty_bst.delete(7)
+    assert empty_bst.root is None
+
+
+def test_delete_bottom_rung(bst_root_fifty):
+    assert bst_root_fifty.depth() == 3
+    for num in [12, 35, 75, 150]:
+        bst_root_fifty.delete(num)
+        assert not bst_root_fifty.contains(num)
+    assert bst_root_fifty.depth() == 2
+    for num in [25, 100]:
+        assert bst_root_fifty._contains(num).left_child is None
+        assert bst_root_fifty._contains(num).right_child is None
+
+
+def test_delete_middle(bst_root_fifty):
+    bst_root_fifty.delete(25)
+    # import pdb; pdb.set_trace()
+    assert bst_root_fifty.root.value == 50
+    assert bst_root_fifty.root.left_child.value == 35 
+    assert bst_root_fifty.root.left_child.left_child is None
+    assert bst_root_fifty.root.left_child.right_child.value == 35
+    assert bst_root_fifty.root.left_child.right_child.parent.value == 12
```

Comment:
Right, we just wanted to assert the parent's connection to the left child was being created properly.